x/concentrated-liquidity/incentives.go

```diff
@@ -213,3 +218,44 @@ func calcAccruedIncentivesForAccum(ctx sdk.Context, accumUptime time.Duration, q
 
 	return incentivesToAddToCurAccum, poolIncentiveRecords, nil
 }
+
+// nolint: unused
+// setIncentiveRecords sets the passed in incentive records in state
+func (k Keeper) setIncentiveRecord(ctx sdk.Context, incentiveRecord types.IncentiveRecord) {
+	// Potentially need to set the pool's incentiveRecord field as well
+	store := ctx.KVStore(k.storeKey)
+	key := types.KeyIncentiveRecord(incentiveRecord.PoolId, incentiveRecord.IncentiveDenom, incentiveRecord.MinUptime)
+	osmoutils.MustSet(store, key, &incentiveRecord)
+}
+
+// nolint: unused
+// setMultipleIncentiveRecords sets multiple incentive records in state
+func (k Keeper) setMultipleIncentiveRecords(ctx sdk.Context, incentiveRecords []types.IncentiveRecord) {
+	for _, incentiveRecord := range incentiveRecords {
+		k.setIncentiveRecord(ctx, incentiveRecord)
+	}
+}
+
+// GetIncentiveRecord gets the incentive record corresponding to the passed in values from store
+func (k Keeper) GetIncentiveRecord(ctx sdk.Context, poolId uint64, denom string, minUptime time.Duration) (types.IncentiveRecord, error) {
+	store := ctx.KVStore(k.storeKey)
+	incentiveStruct := types.IncentiveRecord{}
+	key := types.KeyIncentiveRecord(poolId, denom, minUptime)
+
+	found, err := osmoutils.Get(store, key, &incentiveStruct)
```

Comment:
Does it make sense to use `MustGet` for simplicity because we aren't using the `found` flag for anything other than doing the error check?

I think `Get` and the `found` flag are useful when we need to use the flag for controlling the future execution flow. On the other hand, if we simply return an error when `!found`, we might as well use the `MustGet` instead that encapsulates the error check and panics

Reply:
Hmm I opted for `Get` because I wanted to avoid putting a panic in a getter (this also seems to be a design pattern we follow elsewhere) – did you mean using `MustGet` and then catching the panic to return an error?

Reply:
I see - this sgtm. 

I was coming from the perspective of reducing the code boilerplate. However, your explanation justified keeping it as is